From 3460b450a13fac7a37cd753e6858b18cbb69ff7f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Fri, 9 Aug 2013 11:40:25 +0100
Subject: [PATCH] Resolves: fdo#67743 user autocorr file not written

regression since c9c2fbe485de2bb831bd2e58faef87a19ceff622

revert the logic change and keep the OUString conversion

Change-Id: Ia4f8c3be2b6e325122ae7b21c431c301b8afecad
(cherry picked from commit d2c3297eed9917c110da67b2a4c19265aecb38ed)
---
 editeng/source/misc/svxacorr.cxx | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/editeng/source/misc/svxacorr.cxx b/editeng/source/misc/svxacorr.cxx
index e7a3d31..7be7da1 100644
--- a/editeng/source/misc/svxacorr.cxx
+++ b/editeng/source/misc/svxacorr.cxx
@@ -1545,7 +1545,9 @@ sal_Bool SvxAutoCorrect::CreateLanguageFile( LanguageType eLang, sal_Bool bNewFi
 {
     OSL_ENSURE(pLangTable->find(eLang) == pLangTable->end(), "Language already exists ");
 
-    OUString sShareDirFile( GetAutoCorrFileName( eLang, sal_True, sal_False ));
+    OUString sUserDirFile( GetAutoCorrFileName( eLang, sal_True, sal_False ));
+    OUString sShareDirFile( sUserDirFile );
+
     SvxAutoCorrectLanguageListsPtr pLists = 0;
 
     Time nMinTime( 0, 2 ), nAktTime( Time::SYSTEM ), nLastCheckTime( Time::EMPTY );
@@ -1559,17 +1561,18 @@ sal_Bool SvxAutoCorrect::CreateLanguageFile( LanguageType eLang, sal_Bool bNewFi
         // 2 minutes.
         if( bNewFile )
         {
-            pLists = new SvxAutoCorrectLanguageLists( *this, sShareDirFile, sShareDirFile );
+            sShareDirFile = sUserDirFile;
+            pLists = new SvxAutoCorrectLanguageLists( *this, sShareDirFile, sUserDirFile );
             pLangTable->insert(eLang, pLists);
             aLastFileTable.erase(nFndPos);
         }
     }
-    else if( ( FStatHelper::IsDocument( sShareDirFile ) ||
+    else if( ( FStatHelper::IsDocument( sUserDirFile ) ||
                 FStatHelper::IsDocument( sShareDirFile =
                               GetAutoCorrFileName( eLang, sal_False, sal_False ) ) ) ||
-        ( sShareDirFile = sShareDirFile, bNewFile ))
+        ( sShareDirFile = sUserDirFile, bNewFile ))
     {
-        pLists = new SvxAutoCorrectLanguageLists( *this, sShareDirFile, sShareDirFile );
+        pLists = new SvxAutoCorrectLanguageLists( *this, sShareDirFile, sUserDirFile );
         pLangTable->insert(eLang, pLists);
         if (nFndPos != aLastFileTable.end())
             aLastFileTable.erase(nFndPos);
-- 
1.8.3.1

